### PR TITLE
New nav structure

### PR DESCRIPTION
**Pull request by leejordan**
**Summary**
Creates a new nav structure

**Notable Changes**

Rather than lump everything under 'patterns' and 'layouts' we're now using 'principles', 'basics', 'components', and 'examples'

**Upgrade Steps**

This is only a change to the docs and not the released css or js. However, there may be some broken links or layouts.

**Screenshots**

screen shot 2016-12-12 at 16 27 20

### CODE REVIEW

**Comment by notlee** [comment]
@leejordan looks like some of 'layouts' has survived. A rebase issue?
screen shot 2016-12-13 at 14 32 18

**Comment by notlee** [comment]
The `off-screen-menu` example is quite jarring as it doesn't retain the Pistachio header unlike all the others. Anything we can do to make that better? Perhaps keep the Pistachio header without nav? Something for another PR perhaps.

**Comment by leejordan** [comment]
I am removing that rogue page and I will do something with the off screen menu page shortly.

File: less/modules/off-screen-menu.less [hunk 1]
```diff
@@ -265,7 +265,7 @@
                     .nav__dropdown__menu {
                         visibility: visible;
                         display: block;
-                        max-height: 999px;
+                        max-height: 9999px;
```
Comment (notlee) [inline less/modules/off-screen-menu.less]: Is this max-height for animation? If so increasing it so much could have an impact (change speed). If not, why not the default value `none`?

File: site/docs/views/components_features.hbs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    <li class="grid__col-12">
+                        {{#codeBlock}}
+                            {{{ getFile 'site/docs/views/examples/features/heading.html' }}}
+                        {{/codeBlock}}
+                    </li>
+                </ul>
+            </div>
+
+            <h2 class="cd-text cd-title">feature tile with sash</h2>
+            <p class="cd-text">You can add a <a href="/patterns/sashes">sash</a> to a feature tile.</p>
```
Comment (notlee) [inline site/docs/views/components_features.hbs]: Links should be updated instead of relying on redirects.

**Comment by notlee** [comment]
Also got to update the footer!

File: less/modules/off-screen-menu.less (hunk 1 above)
Comment (leejordan) [inline less/modules/off-screen-menu.less]: true: removed

File: less/docs.less [hunk 3]
```diff
@@ -1,14 +1,6 @@
 @import 'includes/variables';
 @import 'includes/mixins';
 
-body {
```
Comment (leejordan) [inline less/docs.less]: true: I added a div (top border doesn't play nicely with the off screen menu)

File: site/docs/views/examples.hbs [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <div class="container">
     <div class="row">
         <div class="col-xs-12">
-            <h1 class="cd-text cd-title">Layouts</h1>
-            <p class="cd-text">Some layout demonstrations which benefit from a standalone example.</p>
+            <h1 class="cd-text cd-title">Examples</h1>
+            <p class="cd-text">Some examples of components combined into pages.</p>
```
Comment (leejordan) [inline site/docs/views/examples.hbs]: edited... thanks

**Pull request merged**